src/test/crimson/seastore/test_object_data_handler.cc

```diff
@@ -480,6 +540,142 @@ TEST_P(object_data_handler_test_t, multiple_split) {
   });
 }
 
+struct object_data_handler_rbm_overwrite_test_t :
+  object_data_handler_test_t {
+  
+  object_data_handler_rbm_overwrite_test_t() : gen(rd()) {}
+
+  void set_overwrite() {
+    crimson::common::local_conf().set_val("seastore_data_delta_based_overwrite", "131072").get();
+  }
+
+  laddr_t get_random_laddr(size_t block_size, laddr_t limit) {
+    return block_size *
+      std::uniform_int_distribution<>(0, (limit / block_size) - 1)(gen);
+  }
+
+  TransactionRef create_transaction() {
+    return create_mutate_transaction();
+  }
+  std::random_device rd;
+  std::mt19937 gen;
+};
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, multi_write)
+{
+  run_async([this] {
+    set_overwrite();
+    test_multi_write();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, write_hole)
+{
+  run_async([this] {
+    set_overwrite();
+    test_write_hole();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, overwrite_single)
+{
+  run_async([this] {
+    set_overwrite();
+    test_overwrite_single();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, overwrite_double)
+{
+  run_async([this] {
+    set_overwrite();
+    test_overwrite_double();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, overwrite_partial)
+{
+  run_async([this] {
+    set_overwrite();
+    test_overwrite_partial();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, unaligned_write)
+{
+  run_async([this] {
+    set_overwrite();
+    test_unaligned_write();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, unaligned_overwrite)
+{
+  run_async([this] {
+    set_overwrite();
+    test_unaligned_overwrite();
+  });
+}
+
+TEST_P(object_data_handler_rbm_overwrite_test_t, truncate)
+{
+  run_async([this] {
+    set_overwrite();
+    test_truncate();
+  });
+}
```

Comment:
I think it is worth to add the basic tests here because the overwrite is triggered if the write size is less than the threshold value---the above cases works in overwrite mode. We need to verify not only the normal tests (e.g., read, write and unaligned I/Os) but also the split test. How about moving these tests to `object_data_handler_test_t` with renaming.

Reply:
Sounds good to me.